New Entry: C++ Math Functions: `tan()` (#998)
Codecademy/docs: content/cpp/concepts/math-functions/terms/tan/tan.md

---
Title: 'tan()'

> Description: 'Returns the tangent of the argument.'

Reviewer: ```suggestion
Description: 'Returns the tangent of the angle in radians.'
```

Subjects:
  - 'Computer Science'
Tags:
  - 'Functions'
  - 'Arithmetic'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

> The **`tan()`** function returns the tangent of the argument.

Reviewer: ```suggestion
The **`tan()`** function returns the tangent of the angle in radians.
```

## Syntax

```cpp
tan(x)
```

> If the parameter `x` is equal to pi +- n*pi (Where n is a natural number), then a domain error will occur.

Reviewer 2: The asterisk used by itself needs to be escaped or markdown thinks it's an unclosed bold tag.

> If the parameter `x` is equal to pi +- n\*pi (Where n is a natural number), then a domain error will occur.

Reviewer: ```suggestion
The `x` parameter represents an angle measured in radians. One radian is equal to 180/&pi; degrees (where &pi; (Pi) = 3.14159...). If `x` is equal to  &pi; &plusmn; `n`/&pi; (where `n` is a natural number), then a domain error will occur.
```

## Example

> The following example uses the `tan()` function to find the tangent of pi/4:

Reviewer: ```suggestion
The following example uses the `tan()` function to find the tangent of &pi;/4:
```

> The following example uses the `tan()` function to find the tangent of `pi/4`:

Reviewer 2: ```suggestion
The following example uses the `tan()` function to find the tangent of `pi/4`:
```
```suggestion
The following example uses the `tan()` function to find the tangent of pi/4:
```

Reviewer 2: I'd remove the backticks because "pi/4" isn't written how it appears in the code.  You might also mention that `M_PI` is a constant defined in the `cmath` header file.

```cpp
#include <iostream>
#include <cmath>

int main() {
  double x = M_PI/4;
  double result;

  result = std::tan(x);

>   std::cout << "The tangent of " << x << " is " << result << "\n";
>   // Output: The tangent of pi/4 is 1

Reviewer: ```suggestion
  std::cout << "The tangent of " << x << "..." << " is " << result << ".\n";
  // Output: The tangent of 0.785398... is 1.
```

}
```

## Codebyte Example

> The following example is runnable and returns the tangent of `0` with the `tan()` function:

Reviewer 2: I'd think of swapping the two examples and descriptions, so the user is encouraged to play with fractions of pi.

Reviewer: I second this!

```codebyte/cpp
#include <iostream>
#include <cmath>

int main() {
  double x = 0;
  double result;

  result = std::tan(x);

>   std::cout << "The tangent of " << x << " is " << result << "\n";

Reviewer: ```suggestion
  std::cout << "The tangent of " << x << " is " << result << ".\n";
```

---
Title: 'tan()'

> Description: 'Returns the tangent of the angle in radians.'

Reviewer 2: ```suggestion
Description: 'Returns the tangent of an angle given in radians.'
```

Subjects:
  - 'Computer Science'
Tags:
  - 'Functions'
  - 'Arithmetic'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

> The **`tan()`** function returns the tangent of the angle in radians.

Reviewer 2: ```suggestion
The **`tan()`** function returns the tangent of an angle given in radians.
```

## Syntax

> ```cpp

Reviewer 3: ```suggestion
```pseudo
```

tan(x)
```

The `x` parameter represents an angle measured in radians. One radian is equal to 180/&pi; degrees (where &pi; (Pi) = 3.14159...). If `x` is equal to  &pi; &plusmn; `n`/&pi; (where `n` is a natural number), then a domain error will occur.

## Example

> The following example is runnable and returns the tangent of `0` with the `tan()` function:

Reviewer 2: ```suggestion
The following example returns the tangent of `0` with the `tan()` function:
```

```cpp
#include <iostream>
#include <cmath>

int main() {
  double x = 0;
  double result;

  result = std::tan(x);

  std::cout << "The tangent of " << x << " is " << result << ".\n";
    // Output: The tangent of 0 is 0.
}
```

## Codebyte Example

> The following example uses the `tan()` function to find the tangent of &pi;/4:

Reviewer 2: ```suggestion
The following example is runnable and uses the `tan()` function to find the tangent of &pi;/4:
```
